src/ReturnTypes/ModelOnlyDynamicMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Larastan\Larastan\ReturnTypes;
+
+use Illuminate\Database\Eloquent\Model;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\Constant\ConstantArrayTypeBuilder;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\Type;
+
+use function array_map;
+use function array_filter;
+use function count;
+
+final class ModelOnlyDynamicMethodReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+    public function getClass(): string
+    {
+        return Model::class;
+    }
+
+    public function isMethodSupported(MethodReflection $methodReflection): bool
+    {
+        return $methodReflection->getName() === 'only';
+    }
+
+    public function getTypeFromMethodCall(
+        MethodReflection $methodReflection,
+        MethodCall $methodCall,
+        Scope $scope
+    ): Type {
+        $fields = $methodCall->getArgs();
+
+        if (count($fields) < 1) {
+            return new ErrorType();
+        }
+
+        if ($fields[0]->value instanceof Array_) {
+            $fields = $fields[0]->value->items;
+        }
+
+        $fields = array_map(static fn ($f) => $f->value, array_filter($fields));
+
+        $model = $scope->getType($methodCall->var);
+        $array = ConstantArrayTypeBuilder::createEmpty();
+
+        foreach ($fields as $field) {
+            if (! $field instanceof String_) {
+                return new ArrayType(new MixedType(), new MixedType());
+            }
+
+            $name = $field->value;
+
+            if (! $model->hasProperty($name)->yes()) {
+                return new ArrayType(new MixedType(), new MixedType());
```

Comment:
this seems a bit drastic to bail entirely if a single property was not found, as opposed to setting the property to have a `mixed` type or maybe `null`?

Reply:
Yeah, I agree---I'll update it to return null